bin/configure-tower-project.py

```diff
@@ -0,0 +1,455 @@
+                    break
+        # Otherwise, just return their new participant ID for the workspace
+        else:
+            participant_id = response["participant"]["participantId"]
+        return participant_id
+
+    def set_workspace_role(
+        self, org_id: str, ws_id: str, part_id: int, role: str
+    ) -> None:
+        """Update the participant role in the given workspace
+
+        Args:
+            org_id (str): Identifier for the 'Sage Bionetworks' organization
+            ws_id (str): Identifier for the project workspace
+            part_id (int): Participant ID for the user in the given workspace
+            role (str): 'owner', 'admin', 'maintain', 'launch', or 'view'
+        """
+        endpoint = f"/orgs/{org_id}/workspaces/{ws_id}/participants/{part_id}/role"
+        data = {"role": role}
+        self.send_tower_request("PUT", endpoint, json=data)
+
+    def get_all_users(self) -> Iterator:
+        iterator = chain(
+            self.owners, self.admins, self.maintainers, self.launchers, self.viewers
+        )
+        return iterator
+
+    def add_workspace_participants(self, org_id: str, ws_id: str) -> None:
+        """Add maintainers and viewers to the organization and workspace
+
+        Args:
+            org_id (str): Identifier for the 'Sage Bionetworks' organization
+            ws_id (str): Identifier for the project workspace
+        """
+        for email in self.get_all_users():
+            member_id = self.add_organization_member(org_id, email)
+            part_id = self.add_workspace_participant(org_id, ws_id, member_id)
+            role = "maintain" if email in self.maintainers else "view"
+            self.set_workspace_role(org_id, ws_id, part_id, role)
+
+
+def parse_args() -> dict:
+    """Parse and validate command-line arguments
+
+    Returns:
+        dict: Validated command-line arguments
+    """
+    # Parse command-line arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--stack_name", "-s", required=True)
+    parser.add_argument("--owners", "-o", nargs="*", help="List of emails")
+    parser.add_argument("--admins", "-a", nargs="*", help="List of emails")
+    parser.add_argument("--maintainers", "-m", nargs="*", help="List of emails")
+    parser.add_argument("--launchers", "-l", nargs="*", help="List of emails")
+    parser.add_argument("--viewers", "-v", nargs="*", help="List of emails")
+    args = parser.parse_args()
+    # Validate command-line arguments
+    email_regex = re.compile(r"\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b")
+    users = args.owners + args.admins + args.maintainers + args.launchers + args.viewers
+    for user in users:
+        match = email_regex.fullmatch(user)
+        if match is None:
+            raise ValueError(f"'{user}' is not an email")
+    return vars(args)
+
+
+if __name__ == "__main__":
+    args = parse_args()
+    tower = TowerConfigurator(**args)
+    tower.configure()
```

Comment:
A discussion @zaro0508 and I had a week or two ago was whether the after_launch cmd runs every time even if there are no updates. In my case it didn't matter if it did. In your case it could. What I'm imagining is that if this happens then every time we run `sceptre launch prod` this script is going to execute for every project. I doubt we'll have so many projects that this will hammer the api, but it will cause delays. I wonder if we should consider making a separate stack that configures the organization, runs the script once, makes those api calls to check what's in the org, then updates it with anything missing. This may be a case of premature optimization, though.